#! /bin/sh /usr/share/dpatch/dpatch-run
## 50-fix-fd-usage.dpatch by Sven Mueller <debian@incase.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix filedescriptor usage (close STDIN/STDOUT/STDERR if backgrounded)

@DPATCH@
diff -urNad trunk/spampd /tmp/dpep.sr9Ucd/trunk/spampd
--- trunk/spampd	2005-10-18 21:09:45.000000000 +0200
+++ /tmp/dpep.sr9Ucd/trunk/spampd	2005-10-18 21:09:45.686991691 +0200
@@ -881,7 +881,9 @@
 				syslog_ident => 'spampd',
 				syslog_facility => 'mail',
 				background => $background,
-				# setsid => 1,
+ 				setsid => $background,	# this causes stdin/stderr/stdout to
+ 							# be opened to /dev/null, cleanly
+ 							# detaching from current console
 				pid_file => $pidfile,
 				user => $user,
 				group => $group,
